### PR TITLE
HWBuffer master merge resolution attempt #2

### PR DESCRIPTION
**Pull request by dillonhuff**
@jeffsetter this another attempt at resolving merge conflicts. If this branch passes tests on travis we should merge it to master and delete the other pull request.

### CODE REVIEW

**Comment by dillonhuff** [comment]
@jeffsetter are you ok with merging this pull request? If so I am going to merge it.

**Comment by jeffsetter** [comment]
Let me review the changes first.

File: Makefile [hunk 1]
```diff
@@ -2051,16 +2086,12 @@ $(DISTRIB_DIR)/halide.tgz: $(LIB_DIR)/libHalide.a \
 distrib: $(DISTRIB_DIR)/halide.tgz
 
 $(DISTRIB_DIR)/halide_config.make: $(BUILD_DIR)/halide_config.make
-	@-mkdir -p $(@D)
```
Comment (jeffsetter) [inline Makefile:2054]: These lines should not be removed

File: Makefile [hunk 2]
```diff
@@ -61,6 +61,10 @@ LLVM_SYSTEM_LIBS=$(shell ${LLVM_CONFIG} --system-libs --link-static | sed -e 's/
 LLVM_AS = $(LLVM_BINDIR)/llvm-as
 LLVM_NM = $(LLVM_BINDIR)/llvm-nm
 LLVM_CXX_FLAGS = -std=c++11  $(filter-out -O% -g -fomit-frame-pointer -pedantic -W% -W, $(shell $(LLVM_CONFIG) --cxxflags | sed -e 's/\\/\//g' -e 's/\([a-zA-Z]\):/\/\1/g;s/-D/ -D/g;s/-O/ -O/g'))
+#<<<<<<< HEAD
+#LLVM_CXX_FLAGS += -D_GLIBCXX_USE_CXX11_ABI=0
+#=======
+#>>>>>>> upstream/abi_fix
```
Comment (jeffsetter) [inline Makefile]: Many of these commented lines can be removed

File: apps/hardware_benchmarks/hw_support/hardware_image_helpers.h [hunk 3]
```diff
@@ -73,6 +73,7 @@ template <typename T>
 bool compare_images(const Halide::Runtime::Buffer<T>& image0,
                     const Halide::Runtime::Buffer<T>& image1) {
   bool equal_images = true;
+  std::cout << "come to here" <<std::endl;
```
Comment (jeffsetter) [inline apps/hardware_benchmarks/hw_support/hardware_image_helpers.h]: remove

File: apps/hardware_benchmarks/hw_support/hardware_targets.mk [hunk 4]
```diff
@@ -14,7 +14,16 @@ SHELL = bash
 BIN ?= bin
 GOLDEN ?= golden
 HWSUPPORT ?= ../../hw_support
+FUNCUBUF_PATH ?= $(ROOT_DIR)/../../../..
 HALIDE_SRC_PATH ?= ../../../..
+LDFLAGS += -lfuncubuf
+#LDFLAGS += -L$(FUNCUBUF_PATH)/bin -Wl,-rpath,$(FUNCUBUF_PATH) -lfuncubuf
```
Comment (jeffsetter) [inline apps/hardware_benchmarks/hw_support/hardware_targets.mk]: remove commented lines

File: apps/hardware_benchmarks/tests/conv_3_3/output [hunk 5]
```diff
@@ -0,0 +1,9408 @@
+g++ -std=c++11 -I ../../../../distrib/include/ -I ../../../../distrib/tools/ -I /nobackup/joeyliu/aha/coreir/include -L/nobackup/joeyliu/aha/coreir/lib -Wl,-rpath,/nobackup/joeyliu/aha/coreir/lib -I/nobackup/joeyliu/BufferMapping/cfunc/include -g -fno-rtti conv_3_3_generator.cpp ../../../../distrib/lib/libHalide.a ../../../../distrib/tools/GenGen.cpp -o bin/conv_3_3.generator  -ldl -lpthread -lz -lcurses -L/nobackup/joeyliu/aha/coreir/lib -Wl,-rpath,/nobackup/joeyliu/aha/coreir/lib -lcoreir-commonlib -lcoreir -lcoreirsim -lcoreir-float -L/nobackup/joeyliu/BufferMapping/cfunc/bin -lfuncubuf
```
Comment (jeffsetter) [inline apps/hardware_benchmarks/tests/conv_3_3/output]: Remove these output log files

Comment (dillonhuff) [reply apps/hardware_benchmarks/tests/conv_3_3/output]: @jeffsetter removed this log file. Are there others besides the one in conv_3_3?

File: apps/hardware_benchmarks/hw_support/hardware_targets.mk (hunk 4 above)
Comment (dillonhuff) [reply apps/hardware_benchmarks/hw_support/hardware_targets.mk]: Removed those lines.

File: apps/hardware_benchmarks/hw_support/hardware_image_helpers.h (hunk 3 above)
Comment (dillonhuff) [reply apps/hardware_benchmarks/hw_support/hardware_image_helpers.h]: Removed it.

File: Makefile (hunk 2 above)
Comment (dillonhuff) [reply Makefile]: Deleted these git merge resolution comments.

File: Makefile (hunk 1 above)
Comment (dillonhuff) [reply Makefile:2054]: @jeffsetter I did not change these lines in the Makefile, this change appears to have been made in hwbuffer_jeff (which I merged in to this branch a long time ago). In that branches Makefile the ```mkdir -p``` lines do not exist:

https://github.com/StanfordAHA/Halide-to-Hardware/blob/3936fcd232c8d1cba34a4c1318612896682fb760/Makefile#L1964-L1971

Would you still like to re-introduce them?

Comment (jeffsetter) [reply Makefile:2054]: Yes, please re-introduce them.

Comment (dillonhuff) [reply Makefile:2054]: Just reintroduced them.

**Review by jeffsetter — APPROVED** [review]
Changes look good, and it's now passing.

**Pull request merged**